inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -827,3 +829,78 @@ def extract_authors_from_pdf(obj, eng):
+        backup_multi_search = backup_multi_search.add(build_subgroup_collaboration_search(normalized_collaboration_string))
+    return multi_search, backup_multi_search
+
+
+def build_standard_collaboration_search(normalized_collaboration_string):
+    name_search = Q("term", normalized_name_variants={"value": normalized_collaboration_string})
+
+    query = name_search
+    filters_ = Q("exists", field="collaboration")
+    return Search().query(query).filter(filters_).source(['collaboration', 'self', 'legacy_name'])
+
+
+def build_subgroup_collaboration_search(normalized_collaboration_string):
+    subgroup_search = Q("term", normalized_subgroups={"value": normalized_collaboration_string})
+
+    query = subgroup_search
+    filters_ = Q("exists", field="collaboration")
+    return Search().query(query).filter(filters_).source(['collaboration', 'self', 'legacy_name'])
+
+
+@with_debug_logging
+def normalize_collaborations(obj, eng):
+    collaborations = obj.data.get('collaborations', [])
+    if not isinstance(collaborations, list) or len(collaborations) < 1:
+        LOGGER.exception("Metadata are malformed for record %s. Collaborations key is not a list.", obj.id)
+        return obj
+    multi_search, backup_multi_search = prepare_collaboration_multi_search(collaborations)
+    try:
+        search_responses = multi_search.execute()
+        backup_search_responses = backup_multi_search.execute()
+    except ValueError as er:
+        # Ignore normalization on ES errors
+        LOGGER.exception("Cannot perform collaborations normalization due to ES error: %s", er)
+        return obj
+    if not len(search_responses) == len(backup_search_responses):
+        LOGGER.exception("Results count does not match: %s : %s : %s", len(collaborations), len(search_responses), len(backup_search_responses))
+    for idx, response in enumerate(search_responses):
+        if 'record' in obj.data['collaborations'][idx]:
+            continue
+        if not response:
+            # Search in collaboration.subgroup_names
+            response = backup_search_responses[idx]
+            if not response:
+                continue
+        LOGGER.debug("Found: %s ==> %s (%s)", obj.data['collaborations'][idx]['value'].encode('utf-8'), response[0].collaboration.value.encode('utf-8'), response[0].meta.score)
+
+        accelerator_experiment = {"record": response[0].self}
+        if 'legacy_name' in response[0]:
+            accelerator_experiment['legacy_name'] = response[0].legacy_name
+        if "accelerator_experiments" not in obj.data:
+            obj.data['accelerator_experiments'] = []
+
+        if accelerator_experiment not in obj.data['accelerator_experiments']:
+            obj.data['accelerator_experiments'].append(accelerator_experiment)
+
+        obj.data['collaborations'][idx] = {
+            "value": response[0].collaboration.value,
+            "record": response[0].self
+        }
+    return obj
```

Comment:
Needs more `info` logging: what was normalized to what, what failed to be normalized, together with holdingpen ID so curators can inspect this periodically and manually fix issues if needed.

Reply:
:+1: